src/psbt.c

```diff
@@ -1436,6 +1460,36 @@ static int psbt_input_from_tx_input(struct wally_psbt *psbt,
     return ret;
 }
 
+int wally_psbt_add_input_taproot_keypath(struct wally_psbt *psbt,
+                                         uint32_t index, uint32_t flags,
+                                         const unsigned char *pub_key,
+                                         size_t pub_key_len,
+                                         const unsigned char *tapleaf_hashes,
+                                         size_t tapleaf_hashes_len,
+                                         const unsigned char *fingerprint,
+                                         size_t fingerprint_len,
+                                         const uint32_t *child_path,
+                                         size_t child_path_len)
+{
+    if (!psbt_is_valid(psbt) || (psbt->version == PSBT_0 && !psbt->tx) ||
+        index > psbt->num_inputs || flags)
```

Comment:
`>` should be `>=` here. But use the `psbt_get_input`/`psbt_get_output` helpers since they do this checking including the v0 checks for you.